src/main/webapp/histogram.js

```diff
@@ -0,0 +1,253 @@
+import React, { useState } from 'react'
+import { List, Map, Set, fromJS } from 'immutable'
+
+import Autocomplete from '@material-ui/lab/Autocomplete'
+import TextField from '@material-ui/core/TextField'
+import Typography from '@material-ui/core/Typography'
+
+const AttributeSelect = props => {
+  const { attributes, value, onChange } = props
+
+  return (
+    <Autocomplete
+      options={attributes}
+      defaultValue={value}
+      autoSelect
+      disableClearable
+      onChange={(event, value) => {
+        onChange(value)
+      }}
+      renderInput={params => (
+        <TextField {...params} margin="normal" fullWidth />
+      )}
+    />
+  )
+}
+
+const Histogram = props => {
+  const [attribute, setAttribute] = useState(null)
+  const { results } = props
+  const values = fromJS(results || {})
+  const attrSet = values.reduce((acc, data) => {
+    return acc.merge(
+      data
+        .getIn(['metacard', 'properties'])
+        .keySeq()
+        .toSet()
+    )
+  }, Set())
+
+  const filteredResults = values.filter(result => {
+    const attributes = result
+      .getIn(['metacard', 'properties'])
+      .keySeq()
+      .toSet()
+    return attributes.includes(attribute)
+  })
+
+  const attrMap = filteredResults.reduce((acc, result) => {
+    const attributes = result.getIn(['metacard', 'properties'])
+    const value = attributes.get(attribute)
+    return acc.update(value, (val = List()) => val.push(result))
+  }, Map())
+
+  const options = attrSet.toJS()
+
+  return (
+    <div style={{ display: 'flex', flexDirection: 'column' }}>
+      <AttributeSelect
+        attributes={options}
+        value={attribute}
+        onChange={setAttribute}
+      />
+      {attrMap.isEmpty() ? null : (
+        <Graph
```

Comment:
could use props.children in the graph component instead of passing down these as props